teuthology/orchestra/connection.py

```diff
@@ -107,8 +117,14 @@ def connect(user_at_host, host_key=None, keep_alive=False, timeout=60,
                 try:
                     ssh.connect(**connect_args)
                     break
-                except paramiko.AuthenticationException:
-                    log.exception(
-                        "Error connecting to {host}".format(host=host))
+                except RECONNECT_EXCEPTIONS as e:
+                    log.exception(f"Error connecting to {host}: {e}")
+                except Exception as e:
+                    # gevent.__hub_primitives returns a generic Exception, *sigh*
+                    if "timed out" in str(e):
+                        log.exception("Error connecting to {host}: {e}")
```

Comment:
prefix 'f' is missing

I don't see f-strings being used elsewhere in teuthology. So we expect python >=3.6 for testing?

Reply:
This is a core functionality, I would prefer not to produce py3 only code for awhile, we run some tests on py2